weather/stations/davis_weatherLinkIP.py

```diff
@@ -0,0 +1,708 @@
+
+    def _get_loop_fields(self):
+        for i in xrange(3):
+            raw = self._loop_cmd()  # read raw data
+            crc_ok = VProCRC.verify(raw)
+            if crc_ok: break  # exit loop if valid
+            time.sleep(1)
+
+        if not crc_ok:
+            raise NoDeviceException('Can not access weather station')
+
+        return LoopStruct.unpack(raw)
+
+    def _get_new_archive_fields(self):
+        '''
+        returns a dictionary of fields from the newest archive record in the
+        device. return None when no records are new.
+        '''
+        for i in xrange(3):
+            records = self._dmpaft_cmd(self._archive_time)
+            if records is not None: break
+            time.sleep(1)
+
+        if records is None:
+            raise NoDeviceException('Can not access weather station')
+
+        # find the newest record
+        # new_rec = None
+        new_rec = []
+        for r in records:
+            new_time = (r['DateStamp'], r['TimeStamp'])
+            if self._archive_time < new_time:
+                self._archive_time = new_time
+                # new_rec = r
+                new_rec.append(r)
+        return new_rec
+
+    def _calc_derived_fields(self, fields):
+        '''
+        calculates the derived fields (those fields that are calculated)
+        '''
+        # convenience variables for the calculations below
+        temp = fields['TempOut']
+        hum = fields['HumOut']
+        wind = fields['WindAvg']
+        # wind10min = fields['WindSpeed10Min']
+        wind10min = None
+        fields['HeatIndex'] = calc_heat_index(temp, hum)
+        fields['WindChill'] = calc_wind_chill(temp, wind, wind10min)
+        fields['DewPoint'] = calc_dewpoint(temp, hum)
+        fields['DewPointDavis'] = calc_dewpoint_davis(temp, hum)
+        # store current data string
+        now = time.localtime()
+        fields['DateStamp'] = time.strftime("%Y-%m-%d %H:%M:%S", now)
+        # fields['Year'] = now[0]
+        # fields['Month'] = str(now[1]).zfill(2)
+        now = time.gmtime()
+        fields['DateStampUtc'] = time.strftime("%Y-%m-%d %H:%M:%S", now)
+        # fields['YearUtc'] = now[0]
+        # fields['MonthUtc'] = str(now[1]).zfill(2)
+
+    def parse(self):
+        '''
+        read and parse a set of data read from the console.  after the
+        data is parsed it is available in the fields variable.
+        '''
+        # fields = self._get_loop_fields()
+        # fields = {}
+        # fields['Archive'] = self._get_new_archive_fields()
+
+        fields = self._get_new_archive_fields()
+
+        for i in range(0, fields.__len__()):
+            self._calc_derived_fields(fields[i])
+
+        # set the fields variable the the values in the dict
+        self.fields = fields
+
+    def getTime(self):
+        '''
+        Get the current time from the console
+        :returns: the current time in datetime format
+        '''
+        for i in xrange(3):
```

Comment:
well I can imagine... I did it 5 years ago in a python 2.7 project XD. Thanks for the notice... I see if I've time to fix it ASAP. Thanks a lot for your comment

Reply:
:smile:

Reply:
Hey @bgkerion , did you by any chance develop a way to convert measurements into the metric system (celsius instead of farenheit)? If yes, could you point me where into the code?
It sounds trivial but I'm struggling with it!
Thanks a lot in advance!